### ISSUE TITLE
Update table schema doesn't work if there's a null value somewhere in the source statement

### ISSUE THREAD

**Issue by eybert** [TabularEditor/TabularEditor3#282]
This works:
`let
    Source = PowerBI.Dataflows(),
    Workspace = Source{[workspaceId=`
but this doesn't:
`let
    Source = PowerBI.Dataflows(null),
    Workspace = Source{[workspaceId=`
I'm no M expert. The second statement was something PBI Desktop came up with. It's the same thing when there's a null elsewhere in the Source statement as well.

**Comment by otykier** [comment]
That's a bug, thanks for reporting!

**Comment by otykier** [comment]
M parsing of `null` now fixed in [3.1.5](https://github.com/TabularEditor/TabularEditor3/releases/tag/3.1.5). Do note, however, that Tabular Editor 3 doesn't support all Power Query connectors, specifically `PowerBI.Dataflows` is not supported. To work around this, use the `Use Analysis Services for change detection` option. [More information](https://docs.tabulareditor.com/te3/tutorials/importing-tables.html#updating-table-schema-through-analysis-services).

**Comment by otykier** [comment]
For reference, here are the full steps needed in order to add a table from a dataflow:

Dataflows must be added manually, as Microsoft does not provide a way (REST API or otherwise) to obtain schema information from a specific dataflow, from 3rd party applications. As such, to import a table from a dataflow, please follow the steps below (note, this only works while TE3 is connected to the Power BI XMLA endpoint):

1. Add a new table to the model (right-click on “Tables > Create… > Table (Alt+5)”). Make sure the table holds an M Partition (if it uses a Legacy partition, first add an M partition to the table, then delete the Legacy partition).
2. Set the M expression of the partition on the table to the following expression – remember to update the WorkspaceID, DataflowID and EntityName to match your values:

```M
let
    Source = PowerBI.Dataflows(null),
    Workspace = Source{[workspaceId="xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"]}[Data],
    Dataflow = Workspace{[dataflowId="xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"]}[Data],
    Entity = Dataflow{[entity="<EntityName>"]}[Data]
in
    Entity
```

3.	Perform a schema update (right-click on the table, “Update table schema…”). This will let TE3 populate the columns of the table, based on the output of the dataflow.

Note that for step 3 to work, you must enable the **Use Analysis Services for change detection**-option under **Tools > Preferences > Schema compare**. More details about this last point [here](https://docs.tabulareditor.com/te3/tutorials/importing-tables.html#updating-table-schema-through-analysis-services).

Alternatively, a simpler approach may be to use Power BI Desktop to import the tables you need from Dataflows, then open an additional instance of TE3, connect it to the data model in Power BI Desktop and copy over the tables to the model you’re working on.
